### ISSUE TITLE
IUploader is not included in heron-spi.jar

### ISSUE THREAD

**Issue by chris-pardy** [twitter/heron#1130]
The heron spi jar that is built and distributed to Maven Central does not include the com.twitter.heron.spi.uploader package and the IUploader file. These are needed to extend Heron by adding a uploader.

**Comment by lewiskan** [comment]
@chris-pardy @kramasamy Thanks for heads up, the 'heron-spi.jar' artifact uploaded to Maven Central is generated by the Heron release script.  We will need to check the release process to include the IUploader file.  I'm discussing with Karthik to resolve, thanks.

**Comment by nlu90** [comment]
I just looked into the spi-jar genrule, only files in `utils` and `common` are included in the spi jar.

```
java_binary(
    name = "spi-unshaded",
    srcs = glob([
        "**/spi/utils/**/*.java",
        "**/spi/common/**/*.java",
    ]),
    deps = utils_deps_files,
)

genrule(
    name = "heron-spi-jar",
    srcs = [":spi-unshaded_deploy.jar"],
    outs = ["heron-spi.jar"],
    cmd  = "cp $< $@",
)
```

**Comment by kramasamy** [comment]
@chris-pardy - this is fixed in the PR https://github.com/twitter/heron/pull/1133 and it will be available in the next release 0.14.2
